src/core/geometry/qgsabstractgeometry.h

```diff
@@ -484,4 +484,12 @@ struct CORE_EXPORT QgsVertexId
   VertexType type;
 };
 
+#ifndef SIP_RUN
+template <class T>
+inline T qgsgeometry_cast( const QgsAbstractGeometry *geom )
+{
+  return const_cast<T>( reinterpret_cast<T>( 0 )->cast( geom ) );
```

Comment:
Which one?
I took `qobject_cast` as a reference but not completely sure I got everything right.